Runtime/WebViewPanel.cs

```diff
@@ -121,30 +119,42 @@ private void OnWebMessageReceived(string message)
             SDKLogger.AvatarLoaderLogger.Log(TAG, $"--- WebView Message: {message}");
             try
             {
-                var webMessage = JsonConvert.DeserializeObject<WebMessage>(message);
-
-                if (webMessage.eventName == AVATAR_EXPORT_EVENT_NAME)
-                {
-                    if (webMessage.data.TryGetValue(DATA_URL_FIELD_NAME, out var avatarUrl))
-                    {
-                        OnAvatarCreated?.Invoke(avatarUrl);
-
-                        if (urlConfig.clearCache)
-                        {
-                            loaded = false;
-                            webViewObject.Reload();
-                        }
-
-                        SetVisible(false);
-                    }
-                }
+                HandleEvents(JsonConvert.DeserializeObject<WebMessage>(message));
             }
             catch (Exception e)
             {
                 SDKLogger.AvatarLoaderLogger.Log(TAG, $"--- Message is not JSON: {message}\nError Message: {e.Message}");
             }
         }
 
+        private void HandleEvents(WebMessage webMessage)
+        {
+            switch (webMessage.eventName)
+            {
+                case WebViewEventNames.AVATAR_EXPORT:
```

Comment:
I think moving the event here and pre-subbing to it in start would make more sense.

```suggestion
                case WebViewEventNames.AVATAR_EXPORT:
                   OnAvatarCreated?.Invoke(url);
```

Reply:
Hmm what abou this instead? 

```cs

            switch (webMessage.eventName)
            {
                case WebViewEventNames.AVATAR_EXPORT:
                    OnAvatarCreated?.Invoke(webMessage.GetAvatarUrl());
                    HideAndClearCache();
                    break;
....

        private void HideAndClearCache()
        {
            if (urlConfig.clearCache)
            {
                loaded = false;
                webViewObject.Reload();
            }

            SetVisible(false);
        }
```

Reply:
or maybe this 

```cs
                case WebViewEventNames.AVATAR_EXPORT:
                    OnAvatarCreated?.Invoke(webMessage.GetAvatarUrl());
                    HandleClearCache();
                    SetVisible(false);

```

Reply:
Could be, I missed that this event requires func with string param.